lib/web/websocket/websocket.js

```diff
@@ -266,14 +262,10 @@ class WebSocket extends EventTarget {
       // increase the bufferedAmount attribute by the length of the
       // ArrayBuffer in bytes.
 
-      const value = new FastBuffer(data)
-      const frame = new WebsocketFrameSend(value)
-      const buffer = frame.createFrame(opcodes.BINARY)
-
-      this.#bufferedAmount += value.byteLength
-      socket.write(buffer, () => {
-        this.#bufferedAmount -= value.byteLength
-      })
+      this.#bufferedAmount += data.byteLength
+      this.#sendQueue.add(data, () => {
```

Comment:
Will it be changed before waiting for the blob promise?

```js
const ws = new WebSocket("ws://...");
ws.send(new Blob([]));
const buf = new Uint8Array(1)
ws.send(buf);
buf[0] = 1;
```

Reply:
yes, probably. mind writing a test for it?

Reply:
@KhafraDev
```js
const { test } = require("node:test");
const { once } = require("node:events");
const { WebSocketServer } = require("ws");
const { WebSocket } = require("../..");
const { tspl } = require("@matteo.collina/tspl");

test("check cloned", async (t) => {
  const assert = tspl(t, { plan: 2 });

  const server = new WebSocketServer({ port: 0 });

  server.on("connection", (ws) => {
    ws.on("message", (data) => {
        assert.strictEqual(data[0], 0)
    })
  });

  const ws = new WebSocket(`ws://localhost:${server.address().port}`);

  ws.addEventListener("open", () => {
    ws.send(new Blob([new Uint8Array(1)]));
    const buf = new Uint8Array(1);
    ws.send(buf)
    buf[0] = 1;
  });

  t.after(() => {
    server.close()
    ws.close()
  })
  
  await assert.completed
});

```

Reply:
If you create the frame in advance, there is no problem.

Reply:
I opted to clone the frame body rather than prematurely create the frame.